CMake/kwiver-setup-python.cmake

```diff
@@ -0,0 +1,118 @@
+###
+# Finds the python binaries, libraries, inclue, and site-packages paths
+
+
+set(KWIVER_PYTHON_VERSION "2" CACHE STRING "Python version to use: 3 or 2")
+set_property(CACHE KWIVER_PYTHON_VERSION PROPERTY STRINGS "3" "2")
+
+# If we change python versions re-find the bin, include, and libs
+if (NOT _prev_kwiver_python_version STREQUAL KWIVER_PYTHON_VERSION)
+  # but dont clobber initial settings in the instance they are specified via
+  # commandline (e.g  cmake -DPYTHON_EXECUTABLE=/my/special/python)
+  if (_prev_kwiver_python_version)
+    message(STATUS "The Python version changed; refinding the interpreter")
+    unset(PYTHON_EXECUTABLE CACHE)
+    unset(PYTHON_INCLUDE_DIR CACHE)
+    unset(PYTHON_LIBRARY CACHE)
+    unset(PYTHON_LIBRARY_DEBUG CACHE)
+  endif()
+endif()
+
+if (KWIVER_PYTHON_VERSION MATCHES "^3.*")
```

Comment:
If we're using specific version of 2 and 3 here, the above KWIVER_PYTHON_VERSION should accurately reflect these values, e.g. specifically use 2.7 and 3.4 strings, checking for those specific strings here.

Reply:
KWIVER_PYTHON_VERSION  is only allowed to either be the major version; either 2 or 3 (its a combo box). 

I'm open to reworking this versioning code. Its a little weird, but I did it that way to (1) reduce the number changes to the variables, (2) to agree with how its done elsewhere (e.g. fletch, opencv), and (3) to let us default to python 2 and while giving the user the choice of switching to 3.

I think its good to have to have a toggle variable that lets you switch between the two major versions (a-priori without knowing which is available), while having a separate variable to manage the minor ones (perhaps we should mark that variable as advanced?).  On the other hand, we might forgo that and have one variable with the full version. Maybe there's a different way I'm too tired to think of?